Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift

```diff
@@ -153,11 +153,96 @@ struct FrameworkBundleAssembler {
             )
         }
     }
+}
+
+extension FrameworkBundleAssembler {
+    struct ResourcesProcessor {
+        private let isVersionedBundle: Bool
+        private let sourceFrameworkBundlePath: TSCAbsolutePath
+        private let sourceFrameworkInfoPlistPath: TSCAbsolutePath
+        private let sourceResourceBundlePath: TSCAbsolutePath?
+        private let destinationFrameworkBundlePath: TSCAbsolutePath
+        private let fileSystem: any FileSystem
+
+        init(
+            isVersionedBundle: Bool,
+            sourceFrameworkBundlePath: TSCAbsolutePath,
+            sourceFrameworkInfoPlistPath: TSCAbsolutePath,
+            sourceResourceBundlePath: TSCAbsolutePath?,
+            destinationFrameworkBundlePath: TSCAbsolutePath,
+            fileSystem: any FileSystem
+        ) {
+            self.isVersionedBundle = isVersionedBundle
+            self.sourceFrameworkBundlePath = sourceFrameworkBundlePath
+            self.sourceFrameworkInfoPlistPath = sourceFrameworkInfoPlistPath
+            self.sourceResourceBundlePath = sourceResourceBundlePath
+            self.destinationFrameworkBundlePath = destinationFrameworkBundlePath
+            self.fileSystem = fileSystem
+        }
+
+        func copyResources() throws {
```

Comment:
I prefer to pass `destinationFrameworkBundlePath` here instead of holding it as an instant variable.

Reply:
Then let's pass all the parameters could be passed as the arguments for this method 🤔 

- ref: https://github.com/giginet/Scipio/pull/172#discussion_r1913903500

I'm not a fan of defining a subset of `FrameworkComponents`.